regex-syntax/src/parser.rs

```diff
@@ -801,14 +801,25 @@ impl<'a> Chars<'a> {
         }
     }
 
+    /// Skip line comments and whitespace
     fn skip(&mut self) {
         if !self.ignore_space { return; }
         while self.cur < self.chars.len() {
+            // Handle escaping of `#`, i.e. don't start a comment with `\#`.
+            if !self.in_comment && (self.c() == '\\')
+                && ((self.cur + 1) < self.chars.len())
+                && (self.chars[self.cur + 1] == '#')
```

Comment:
Your have your overflow paranoia, I have my parens. ;)

How much can I get rid of? Does `+` bind stronger than `<` which binds stronger than `&&`?